File: algorithms/math/newtons-method/swift/IntegerSqrt.swift
func integerSqrt(_ arr: [Int]) -> Int {
    let n = arr[0]
    if n <= 1 { return n }
    var x = n
    while true {
        let x1 = (x + n / x) / 2
        if x1 >= x { return x }
        x = x1
    }
}

File: algorithms/math/newtons-method/python/integer_sqrt.py
def integer_sqrt(arr: list[int]) -> int:
    n = arr[0]
    if n <= 1:
        return n
    x = n
    while True:
        x1 = (x + n // x) // 2
        if x1 >= x:
            return x
        x = x1
